forge/test/mlir/test_training.py

```diff
@@ -11,17 +11,18 @@
 from forge.verify.compare import compare_with_golden
 
 
-@pytest.mark.push
-def test_torch_training():
-    class MatmulParam(nn.Module):
-        def __init__(self):
-            super().__init__()
-            self.p = nn.Parameter(torch.rand(1024, 1024))
-            nn.init.xavier_uniform_(self.p)
+class MatmulParam(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.p = nn.Parameter(torch.rand(1024, 1024))
+        nn.init.xavier_uniform_(self.p)
+
+    def forward(self, x):
+        return torch.matmul(x, self.p)
 
-        def forward(self, x):
-            return torch.matmul(x, self.p)
 
+@pytest.mark.push
+def test_torch_training():
```

Comment:
Should we update this test to run the optimizer on the device as well? 

Even better, I would suggest to have:
1. One test where loss and optimizer are no host
2. Second test where loss is on device, optimizer on host
3. Third test where both loss and optimizer are on device

This is good showcase how to manage each case, and can be helpful for future debugging.

Reply:
I agree, and we can have the 4th option as well (loss on host, opt on device)

Reply:
This sounds like a good task for training team.